### PR TITLE
Add webservice to request latest PkPass

### PR DESCRIPTION
**Pull request by ueman**
Unfortunately, I don't have a pass to test this functionality, but based on the docs, this should work.

I'll tackle todos and other parts of the web at some other time.

Partially adresses https://github.com/ueman/passkit/issues/9

### CODE REVIEW

File: passkit/lib/src/passkit_web_client.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      case 304:
+        return null;
+      case 401:
+        // TODO: This should probably be a custom exception
+        throw Exception('Authorization error');
+      default:
+        // TODO: This should probably be a custom exception
+        throw Exception(
+            'Unrecognized status code returned: ${response.statusCode}');
+    }
```
Comment (marcossevilla) [inline passkit/lib/src/passkit_web_client.dart:68]:
```suggestion
    return switch (response.statusCode) {
      200 => response.bodyBytes,
      304 => null,
      // TODO: This should probably be a custom exception
      401 => throw Exception('Authorization error'),
      // TODO: This should probably be a custom exception
      _ => throw Exception('Unrecognized status code returned: ${response.statusCode}'),
    };
```

File: passkit/lib/src/passkit_web_client.dart [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    final endpoint =
+        Uri.parse(webServiceUrl).resolveUri(Uri(pathSegments: ['v1', 'log']));
+
+    await _client.post(endpoint, body: jsonEncode(messages));
+  }
+}
+
+class PassWebServiceUnsupported implements Exception {}
```
Comment (marcossevilla) [inline passkit/lib/src/passkit_web_client.dart:88]: can we have all exceptions on a different file? at least related ones imo

Comment (ueman) [reply passkit/lib/src/passkit_web_client.dart:88]: Sure

Comment (ueman) [reply passkit/lib/src/passkit_web_client.dart]: Adressed ✅

File: passkit/lib/src/passkit/pkpass.dart [hunk 3]
```diff
@@ -168,4 +168,6 @@ class PkPass {
   // final Uri? passURL;
 
   final List<int> sourceData;
+
+  bool isWebServiceAvailable() => pass.webServiceURL != null;
```
[the remark this answers is not in the document]
Comment (ueman) [reply passkit/lib/src/passkit/pkpass.dart]: Adressed ✅

**Review by marcossevilla — APPROVED** [review]

**Pull request merged**